torchvision/io/video_reader.py

```diff
@@ -105,7 +125,24 @@ def __init__(self, path: str, stream: str = "video", num_threads: int = 0, devic
                 + "build torchvision from source."
             )
 
-        self._c = torch.classes.torchvision.Video(path, stream, num_threads)
+        if src == "":
+            if path is None:
+                raise ValueError("src cannot be empty")
```

Comment:
NIT: I just realized that missing argument should raise `TypeError` instead of `ValueError`. Can we change this to `TypeError` instead?

Reply:
In this case I disagree, we are checking for the value of the variable, not for the type. Would like to understand your interpretation in case I am missing something.

Reply:
I just tried on my local python the following:
```
>>> def foo(x, y):
...  print(x, y)
... 
>>> foo("hello")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: foo() missing 1 required positional argument: 'y'
```

Notice that when we dont specify 1 required positional argument, python raise `TypeError` instead of `ValueError` (I actually surprised that python use `TypeError` for this). Therefore I think we should follow this convention since in this case we basically missing 1 required argument as well.